CONTRIBUTING.md

```diff
@@ -210,6 +210,7 @@ General guidelines for pull requests (PRs):
 * Larger changes should be broken down into smaller components and integrated
   separately.
 * Bug fixes should be submitted in separate PRs.
+* New files should use a hyphen between words in the file name, not an underscore.
```

Comment:
This seems rather broad in scope. Do we want to restrict this to just the 'doc/' and 'examples/' folder, or include things in '.github/' and 'pygmt/' and the top level directory as well? I don't think it's ok to have hyphens for *.py files that have to be imported for example (e.g. `import pygmt.datasets.earth-relief` won't work on an `earth-relief.py` I think). I think underscores are actually preferred in most Python packages (e.g. https://pyphilly.org/hyphen-and-underscores-when-use-each-django-and-python/).

Reply:
Apologies @willschlitzer, I think we should stick with underscore. I just had a look around the galleries of `matplotlib`, `cartopy` and `geopandas` and they all use underscores:

- https://github.com/matplotlib/matplotlib/tree/v3.3.4/examples/pyplots
- https://github.com/SciTools/cartopy/tree/v0.18.0/lib/cartopy/examples
- https://github.com/geopandas/geopandas/tree/v0.9.0/doc/source/gallery

To be honest, I think we can close this PR #1034 and #1024 too since it's a Python standard in PEP8. Really sorry that I didn't catch this earlier yesterday, I know this must have taken up a lot of your time :bow: :bow: :bow:

Reply:
I agree that underscores are better than hyphens for Python scripts. Sorry for the inconvenience.

Reply:
BTW, we should still add a guideline like "Use underscores for all python file names, not hyphens".

Reply:
If we use underscores, shall we also make it as a rule in contributing.md?or, we assume everyone know it. @weiji14

Reply:
Sure, let's put this 'underscore' rule in the guideline and also hint at [PEP8](https://www.python.org/dev/peps/pep-0008/).

```suggestion
* Use underscore for all Python (*.py) files as per [PEP8](https://www.python.org/dev/peps/pep-0008/), not hyphens
```